### ISSUE TITLE
fails to run jekyll locally

### ISSUE THREAD

**Issue by fkanehiro** [jrl-umi3218/jrl-umi3218.github.com#21]
I could run jekyll locally until 4187d6d6e3ee5cd08c634ff48eb1637eb1e5921b but it fails with the following error message since 332d67672b3591161e4d08f14a486d5b58695f8a.
```
jekyll server --safe --trace
Configuration file: /home/jenkinshrg/src/jrl-umi3218.github.com/_config.yml
       Deprecation: The 'plugins' configurationoption has been renamed to 'plugins_dir'. Please update your config file accordingly.
            Source: /home/jenkinshrg/src/jrl-umi3218.github.com
       Destination: /home/jenkinshrg/src/jrl-umi3218.github.com/_site
 Incremental build: disabled. Enable with --incremental
      Generating... 
  Liquid Exception: Liquid syntax error (line 84): Tag '{%- if page.title -%}' was not properly terminated with regexp: /\%\}/ in _layouts/default.html
Traceback (most recent call last):
	24: from /usr/bin/jekyll:11:in `<main>'
	23: from /usr/lib/ruby/vendor_ruby/mercenary.rb:19:in `program'
	22: from /usr/lib/ruby/vendor_ruby/mercenary/program.rb:42:in `go'
	21: from /usr/lib/ruby/vendor_ruby/mercenary/command.rb:220:in `execute'
	20: from /usr/lib/ruby/vendor_ruby/mercenary/command.rb:220:in `each'
	19: from /usr/lib/ruby/vendor_ruby/mercenary/command.rb:220:in `block in execute'
	18: from /usr/lib/ruby/vendor_ruby/jekyll/commands/serve.rb:34:in `block (2 levels) in init_with_program'
	17: from /usr/lib/ruby/vendor_ruby/jekyll/commands/build.rb:33:in `process'
	16: from /usr/lib/ruby/vendor_ruby/jekyll/commands/build.rb:60:in `build'
	15: from /usr/lib/ruby/vendor_ruby/jekyll/command.rb:26:in `process_site'
	14: from /usr/lib/ruby/vendor_ruby/jekyll/site.rb:59:in `process'
	13: from /usr/lib/ruby/vendor_ruby/jekyll/site.rb:177:in `render'
	12: from /usr/lib/ruby/vendor_ruby/jekyll/site.rb:177:in `each'
	11: from /usr/lib/ruby/vendor_ruby/jekyll/site.rb:179:in `block in render'
	10: from /usr/lib/ruby/vendor_ruby/jekyll/renderer.rb:70:in `run'
	 9: from /usr/lib/ruby/vendor_ruby/jekyll/renderer.rb:146:in `place_in_layouts'
	 8: from /usr/lib/ruby/vendor_ruby/jekyll/renderer.rb:106:in `render_liquid'
	 7: from /usr/lib/ruby/vendor_ruby/jekyll/liquid_renderer/file.rb:10:in `parse'
	 6: from /usr/lib/ruby/vendor_ruby/jekyll/liquid_renderer/file.rb:43:in `measure_time'
	 5: from /usr/lib/ruby/vendor_ruby/jekyll/liquid_renderer/file.rb:11:in `block in parse'
	 4: from /usr/lib/ruby/vendor_ruby/liquid/template.rb:108:in `parse'
	 3: from /usr/lib/ruby/vendor_ruby/liquid/template.rb:122:in `parse'
	 2: from /usr/lib/ruby/vendor_ruby/liquid/document.rb:5:in `parse'
	 1: from /usr/lib/ruby/vendor_ruby/liquid/tag.rb:10:in `parse'
/usr/lib/ruby/vendor_ruby/liquid/block.rb:41:in `parse': Liquid syntax error (line 84): Tag '{%- if page.title -%}' was not properly terminated with regexp: /\%\}/ (Liquid::SyntaxError)

```
I'm using ubuntu18.04 and jekyll installed by apt (3.1.6).

**Comment by aescande** [comment]
I had to use a recent [whitespace control](https://shopify.github.io/liquid/basics/whitespace/) syntax to avoid the generation of tens of thousands of white line in many files, and this requires liquid 4.0, and thus a newer version of Jekyll.

The best way is to install and update jekyll as presented [here](https://jekyllrb.com/docs/installation/ubuntu/). You'll have to remove the apt installation of jekyll first including an `apt-get autoremove`. It should then work using `bundle exec jekyll serve --trace`

**Comment by mehdi-benallegue** [comment]
This issue is related to #19

I updated the instructions in the README file to better reflecting the appropriate steps.

**Comment by fkanehiro** [comment]
@aescande @mehdi-benallegue 
Thank you for your comments.
I followed the instructions but I got the following messages.
```
$ bundle exec jekyll server --safe --trace
Traceback (most recent call last):
	2: from /home/jenkinshrg/gems/bin/bundle:23:in `<main>'
	1: from /usr/lib/ruby/2.5.0/rubygems.rb:308:in `activate_bin_path'
/usr/lib/ruby/2.5.0/rubygems.rb:289:in `find_spec_for_exe': can't find gem bundler (>= 0.a) with executable bundle (Gem::GemNotFoundException)
```
I'm sure that I installed bundler.
```
$ gem install jekyll bundler
Successfully installed jekyll-4.2.0
Parsing documentation for jekyll-4.2.0
Done installing documentation for jekyll after 0 seconds
Successfully installed bundler-2.2.15
Parsing documentation for bundler-2.2.15
Done installing documentation for bundler after 2 seconds
2 gems installed
```
For the meanwhile, I'm using the method using Docker.


```

**Comment by mehdi-benallegue** [comment]
I think it can be fixed by running  `bundle install`. 

If it is the case I will add it to the README

**Comment by fkanehiro** [comment]
I found that the error was caused by version mismatch of bundler.
gem installed bundler 2.2.15 but Gemfile.lock was expecting 2.1.4.
The problem was solved by installing bundler 2.1.4.
I'm not sure why this issue happened only in my environment.

https://qiita.com/mogurinchu/items/9c87a6f16de0b8f7d531 (Japanese)
